### PR TITLE
refactor: extract UI and reporting

### PR DESCRIPTION
**Pull request by alanshaw**
This PR extracts the UI and reporting elements from `run-node.go`. This is an exercise in separating concerns so that tests can be written.

The gist is that you create a `Reporter` which is responsible for collecting reports and periodically publishing them to a channel. The UI consumes the reports and displays them on screen.

We now have some basic tests for reporting and the UI \o/

TODO: the runners need to return a handle that we can use to stop everything. This will make them testable (I'll send a separate PR for this!).

### CODE REVIEW

File: httpapi/httpapi_test.go [hunk 1]
```diff
@@ -7,33 +7,12 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/ipfs/go-datastore"
 	"github.com/libp2p/go-libp2p-core/peer"
-	"github.com/libp2p/hydra-booster/node"
-	"github.com/libp2p/hydra-booster/opts"
-	"github.com/multiformats/go-multiaddr"
+	hytesting "github.com/libp2p/hydra-booster/testing"
```
Comment (daviddias) [inline httpapi/httpapi_test.go]: Why `testing` and not `test?

File: testing/helpers.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// SpawnNode a new Hydra nodes with an in memory datastore and 0 bootstrap peers by default.
+// It also waits for bootstrapping to complete.
+func SpawnNode(options ...opts.Option) (*node.HydraNode, error) {
+	nd, bsCh, err := node.NewHydraNode(append(defaults, options...)...)
+	if err != nil {
+		return nil, err
+	}
+
+	for {
+		status, ok := <-bsCh
```
Comment (daviddias) [inline testing/helpers.go]: Asked to the group if for single notifications, if the best is channels or waitgroups. Not blocking the PR. will soon learn from the maaastersss :)

File: httpapi/httpapi_test.go (hunk 1 above)
Comment (alanshaw) [reply httpapi/httpapi_test.go]: Because the golang module is called "testing" and I read a recommendation on a blog to call it "testing".

Comment (daviddias) [reply httpapi/httpapi_test.go]: Good to know! thanks

**Pull request merged**